projects/packages/my-jetpack/_inc/data/use-jetpack-api-query.ts

```diff
@@ -0,0 +1,36 @@
+import restApi from '@automattic/jetpack-api';
+import { useQuery } from '@tanstack/react-query';
+import { useFetchingErrorNotice } from './notices/use-fetching-error-notice';
+
+/**
+ * A hook to fetch data from the Jetpack API using react-query
+ *
+ * @param {string} name - The name of the query.
+ * @param {Function} queryFn - The function that fetches the data.
+ * @param {string} explicitKey - An optional key to use for the query cache.
+ * @returns {Array} The result of the query.
+ */
+const useJetpackApiQuery = < T >(
+	name: string,
+	queryFn: ( api: typeof restApi ) => Promise< T >,
+	explicitKey?: string
+): ReturnType< typeof useQuery > => {
+	const queryResult = useQuery( {
+		queryKey: [ name, explicitKey ],
+		queryFn: () => {
+			const { apiRoot, apiNonce } = window?.myJetpackRest || {};
+			restApi.setApiRoot( apiRoot );
+			restApi.setApiNonce( apiNonce );
+			return queryFn( restApi );
+		},
+		refetchOnWindowFocus: false,
+		refetchIntervalInBackground: false,
+	} );
+
+	const { isError, isLoading } = queryResult;
+	useFetchingErrorNotice( name, ! isLoading && isError );
```

Comment:
Might want to set an error message override for future use, not required by I'm adding it to the simple query and mutation one so it might be good to add it here too

Reply:
I'll leave to-do note to tackle that during cleanup, so not to double your work that you've already done in your PR.

Reply:
I'm not sure I completely understand, why would this increase work in my PR? The `useFetchingErrorNotice` already accepts the message override, we'd just have to add it as an optional prop for `useJetpackApiQuery` and pass it to the hook

Reply:
The `useFetchingErrorNotice` hook already accepts an override message, I think it would be pretty easy to add it here without affecting my PR at all 🤔

Reply:
Ah true, we just change there the way we accept params, adding this to the PR and will resolve the conflict when merging